Added handler for manifest.json

samdutton opened the pull request:
See #49.

chuckhays commented on src/app_engine/app.yaml:
```diff
@@ -9,6 +9,14 @@ handlers:
 
+- url: /(.*\.json)
+  static_files: html/\1
+  upload: html/(.*\.json)
```
This makes me nervous because we are storing things like the gcm server key in json files. Could we add a specific file name here?

jiayliu replied:
The key files should not be in the html/ folder. But I agree we should be
more specific just to be safe.

On Thu, Feb 12, 2015 at 9:20 AM, Chuck Hays <notifications@github.com>
wrote:

> In src/app_engine/app.yaml
> <https://github.com/webrtc/apprtc/pull/66#discussion_r24598888>:
>
> > @@ -9,6 +9,14 @@ handlers:
> >    static_files: html/\1
> >    upload: html/(.*\.html)
> >
> > +- url: /(.*\.json)
> > +  static_files: html/\1
> > +  upload: html/(.*\.json)
>
> This makes me nervous because we are storing things like the gcm server
> key in json files. Could we add a specific file name here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/webrtc/apprtc/pull/66/files#r24598888>.
>



-- 
Jiayang Liu

samdutton replied:
Good point. Changed in next commit.

chuckhays commented:
LGTM. Thanks for indulging my paranoia.

samdutton commented:
I think this is sensible, not paranoid :).

On 12 February 2015 at 17:38, Chuck Hays <notifications@github.com> wrote:

> LGTM. Thanks for indulging my paranoia.
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/webrtc/apprtc/pull/66#issuecomment-74115515>.
>

samdutton commented:
@chuckhays any idea why the test module imports are failing, as reported by Travis?

jiayliu commented:
Sam,

That's not related to your code. I've reverted the offending PR.

On Fri, Feb 20, 2015 at 7:50 AM, Sam Dutton <notifications@github.com>
wrote:

> @chuckhays <https://github.com/chuckhays> any idea why the test module
> imports are failing, as reported by Travis?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/webrtc/apprtc/pull/66#issuecomment-75259988>.
>



-- 
Jiayang Liu

samdutton commented:
Fair point :).

I made the apprtc.py push without realising I hadn't yet merged the
manifest stuff...

Will try again.

On 20 February 2015 at 17:28, Jiayang Liu <notifications@github.com> wrote:

> Sam,
>
> That's not related to your code. I've reverted the offending PR.
>
> On Fri, Feb 20, 2015 at 7:50 AM, Sam Dutton <notifications@github.com>
> wrote:
>
> > @chuckhays <https://github.com/chuckhays> any idea why the test module
> > imports are failing, as reported by Travis?
> >
> > --
> > Reply to this email directly or view it on GitHub
> > <https://github.com/webrtc/apprtc/pull/66#issuecomment-75259988>.
> >
>
>
>
> --
> Jiayang Liu
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/webrtc/apprtc/pull/66#issuecomment-75281496>.
>

jiayliu commented:
Do you mean pushing to appspot.com? Could you coordinate with Vikas?

On Fri, Feb 20, 2015 at 9:31 AM, Sam Dutton <notifications@github.com>
wrote:

> Fair point :).
>
> I made the apprtc.py push without realising I hadn't yet merged the
> manifest stuff...
>
> Will try again.
>
> On 20 February 2015 at 17:28, Jiayang Liu <notifications@github.com>
> wrote:
>
> > Sam,
> >
> > That's not related to your code. I've reverted the offending PR.
> >
> > On Fri, Feb 20, 2015 at 7:50 AM, Sam Dutton <notifications@github.com>
> > wrote:
> >
> > > @chuckhays <https://github.com/chuckhays> any idea why the test module
> > > imports are failing, as reported by Travis?
> > >
> > > --
> > > Reply to this email directly or view it on GitHub
> > > <https://github.com/webrtc/apprtc/pull/66#issuecomment-75259988>.
> > >
> >
> >
> >
> > --
> > Jiayang Liu
> >
> > --
> > Reply to this email directly or view it on GitHub
> > <https://github.com/webrtc/apprtc/pull/66#issuecomment-75281496>.
>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/webrtc/apprtc/pull/66#issuecomment-75282786>.
>



-- 
Jiayang Liu

samdutton commented:
Sorry - didn't read your email properly - I thought you were referring to
the fact that I added a commit (the one that tweaks apprtc.py) that didn't
relate to the rest of my pull request.

Ignore me :).

On 20 February 2015 at 17:55, Jiayang Liu <notifications@github.com> wrote:

> Do you mean pushing to appspot.com? Could you coordinate with Vikas?
>
> On Fri, Feb 20, 2015 at 9:31 AM, Sam Dutton <notifications@github.com>
> wrote:
>
> > Fair point :).
> >
> > I made the apprtc.py push without realising I hadn't yet merged the
> > manifest stuff...
> >
> > Will try again.
> >
> > On 20 February 2015 at 17:28, Jiayang Liu <notifications@github.com>
> > wrote:
> >
> > > Sam,
> > >
> > > That's not related to your code. I've reverted the offending PR.
> > >
> > > On Fri, Feb 20, 2015 at 7:50 AM, Sam Dutton <notifications@github.com>
> > > wrote:
> > >
> > > > @chuckhays <https://github.com/chuckhays> any idea why the test
> module
> > > > imports are failing, as reported by Travis?
> > > >
> > > > --
> > > > Reply to this email directly or view it on GitHub
> > > > <https://github.com/webrtc/apprtc/pull/66#issuecomment-75259988>.
> > > >
> > >
> > >
> > >
> > > --
> > > Jiayang Liu
> > >
> > > --
> > > Reply to this email directly or view it on GitHub
> > > <https://github.com/webrtc/apprtc/pull/66#issuecomment-75281496>.
> >
> > >
> >
> > --
> > Reply to this email directly or view it on GitHub
> > <https://github.com/webrtc/apprtc/pull/66#issuecomment-75282786>.
>
> >
>
>
>
> --
> Jiayang Liu
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/webrtc/apprtc/pull/66#issuecomment-75287309>.
>

samdutton commented:
Added keyboard shortcut icon toggle as per #157.

samdutton commented:
I've included code in this PR to redirect from apprtc.net to www.apprtc.net, to avoid problems with https://apprtc.net – but is there a way to handle this?

@jiayliu @chuckhays

chuckhays commented:
LGTM. I don't know about the www vs non.

KaptenJansson commented:
What's left to do here?

KaptenJansson commented:
LGTM, @samdutton anything left to do here?

KaptenJansson commented:
Aha this is probably the cause of  #181 as well, it seems to trigger the "Add to homescreen" logic in other ways than using Add to homescreen on an Android device (seen it on desktop as well).

samdutton commented:
LGTM.

KaptenJansson commented:
Will push to GAE tomorrow
